src/codemodder/__main__.py

```diff
@@ -54,13 +40,16 @@ def run_codemods_for_file(
         wrapper = cst.MetadataWrapper(source_tree)
         codemod = codemod_kls(
             CodemodContext(wrapper=wrapper),
+            # TODO: eventually pass execution context here
+            # It will be used for things like dependency management
             file_context,
         )
         if not codemod.should_transform:
             continue
 
         logger.info("Running codemod %s for %s", name, file_context.file_path)
         output_tree = codemod.transform_module(source_tree)
+        # TODO: there has got to be a more efficient way to check this?
```

Comment:
Couldn't we set some kind of flag on the `FileContext` from within the codemod itself?

Reply:
If we want to be reductionist, we could simply check if any changes were added to the changeset. However that would fail to catch codemods that make changes but don't report. Of course, those can be considered bugs, but those happen somewhat often.